syntax = "proto3";

package API.RPC;
import "ciri/api/grpc/proto/messages.proto";

service API {
 rpc addNeighbors (addNeighborsRequest) returns (addNeighborsResponse);

 rpc attachToTangle (attachToTangleRequest) returns (attachToTangleResponse);

 rpc broadcastTransactions (broadcastTransactionsRequest) returns (broadcastTransactionsResponse);

 rpc checkConsistency (checkConsistencyRequest) returns (checkConsistencyResponse);

 rpc findTransactions (findTransactionsRequest) returns (findTransactionsResponse);

 rpc getBalances (getBalancesRequest) returns (getBalancesResponse);

 rpc getInclusionStates (getInclusionStatesRequest) returns (getInclusionStatesResponse);

 rpc getMissingTransactions (getMissingTransactionsRequest) returns (getMissingTransactionsResponse);

 rpc getNeighbors (getNeighborsRequest) returns (getNeighborsResponse);

 rpc getNodeInfo (getNodeInfoRequest) returns (getNodeInfoResponse);

 rpc getTips (getTipsRequest) returns (getTipsResponse);

 rpc getTransactionsToApprove (getTransactionsToApproveRequest) returns (getTransactionsToApproveResponse);

 rpc getTrytes (getTrytesRequest) returns (getTrytesResponse);

 rpc interruptAttachingToTangle (interruptAttachingToTangleRequest) returns (interruptAttachingToTangleResponse);

 rpc removeNeighbors (removeNeighborsRequest) returns (removeNeighborsResponse);

 rpc storeTransactions (storeTransactionsRequest) returns (storeTransactionsResponse);

 rpc wereAddressesSpentFrom (wereAddressesSpentFromRequest) returns (wereAddressesSpentFromResponse);
}
